crypto/objects/o_names.c

```diff
@@ -102,8 +102,15 @@ int OBJ_NAME_new_index(unsigned long (*hash_func) (const char *),
         name_funcs->hash_func = OPENSSL_LH_strhash;
         name_funcs->cmp_func = obj_strcmp;
         CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_DISABLE);
-        sk_NAME_FUNCS_push(name_funcs_stack, name_funcs);
+
+        ret = sk_NAME_FUNCS_push(name_funcs_stack, name_funcs);
```

Comment:
Using `ret` here overwrites the previous assignment, which seems to change the return value behaviour of this function.